File: 0048-rotate-image/0048-rotate-image.java
class Solution {
    public void rotate(int[][] matrix) {
        int c = matrix.length;
        int[][] arr = new int[c][c];
        int b = matrix.length - 1;
        for(int i = 0; i < c; i++){
            for(int j = 0; j < c; j++){
                arr[i][j] = matrix[i][j];
            }
        }
        for(int i = 0; i < c; i++){
            for(int j = 0; j < c; j++){
                matrix[i][j] = arr[b - j][i];
            }
        }
    }
}

File: 0048-rotate-image/0048-rotate-image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int a = 0 , b = matrix.size() - 1;
        int c = matrix.size();
        vector<vector<int>>vec(c , vector<int>(c , 0));
        for(int i = 0; i < matrix.size(); i++){
            for(int j = 0; j < matrix[i].size(); j++){
                vec[i][j] = matrix[i][j];
            }
        }
        for(int i = 0; i < vec.size(); i++){
            for(int j = 0; j < vec[i].size(); j++){
                matrix[i][j] = vec[b - j][i];
            }
            cout << endl;
        }
        
    }
};